### PR TITLE
Handle struct fields when marshalling in json api package

### PR DESCRIPTION
**Pull request by arnold-iakab**

### CODE REVIEW

File: http/jsonapi/response.go [hunk 1]
```diff
@@ -317,6 +316,26 @@ func visitModelNode(model interface{}, included *map[string]*Node,
 					return nil, fmt.Errorf("decimal.MarshalJSONWithoutQuotes needs to be turned on to export decimals as numbers")
 				}
 				node.Attributes[args[1]] = json.RawMessage(d.String())
+			} else if fieldValue.Type() == reflect.TypeOf(new(decimal.Decimal)) {
```
Comment (pieceofsoul) [inline http/jsonapi/response.go:319]: These changes should be covered by unit tests as well.

**Comment by pieceofsoul** [comment]
Looking good to me on the first look. I would propose to improve the added tests a little bit.

**Comment by daemonfire300** [comment]
> Looking good to me on the first look. I would propose to improve the added tests a little bit.

Agree

File: http/jsonapi/response.go (hunk 1 above)
Comment (arnold-iakab) [reply http/jsonapi/response.go:319]: Covered the changes with unit tests. Vulnerability scan is killing the CI tests again :face_with_spiral_eyes:

**Review by Ferlonas — APPROVED** [review]
lgtm

**Comment by daemonfire300** [comment]
Lgtm

**Review by daemonfire300 — APPROVED** [review]

**Pull request merged**